pkg/sql/scanner/scan.go

```diff
@@ -697,9 +697,13 @@ func (s *Scanner) scanNumber(lval ScanSymType, ch int) {
 		// string as an octal literal. Note: we can't use strings.TrimLeft
 		// here, because it will truncate '0' to ''.
 		if !isHex {
-			for len(lval.Str()) > 1 && lval.Str()[0] == '0' {
-				lval.SetStr(lval.Str()[1:])
+			i, err := strconv.ParseInt(lval.Str(), 10, 64)
```

Comment:
looks like we can't do this because `9223372036854775808` and (`-(9223372036854775808)`) isn't handled correctly (it overflows int64). let's revert it.